Refactor sw

helo9 opened the pull request:
Refactor software for simpler extension with more functionality.

helo9 committed:
[SW] Create docker environment for building

helo9 committed:
[SW] Move TCA driver into separate file

helo9 committed:
[SW] Adapt code formatting

helo9 committed:
[SW]  Move total pressure code into separate file

helo9 committed:
[SW] Move ELV sensor code into separate file

helo9 committed:
[SW] Refine Build instructions

helo9 committed:
[SW] Adapt style

AndreasWes commented on software/BPS120.h:
```diff
@@ -0,0 +1,25 @@
+
+class BPS120 {
+  public:
+    BPS120(TwoWire *i2c, float factor, float offset); 
```
inconsistent compared to AllSensorsELV

helo9 replied:
adapted

helo9 committed:
[SW] Move presence check into sensor driver

helo9 committed:
[SW] add preliminary sdcard logging

helo9 committed:
[SW] Improve SD Card Logging

Previously serial console was redirected into log file,
now a proper CSV file is generated.

helo9 committed:
[SW] Move logic from loop to FreeRTOS Task

This allows us to have parallel measurement and storing tasks.

helo9 committed:
[SW] Move SD card writing into task

helo9 committed:
[SW] Add status indication task

The Status indication task tracks the health of other
Tasks and manipulates the board LED. It is blinking,
when operational and static if not.

helo9 committed:
[SW] Introduce generic printStatus method

Each task can implement a `printStatus` method to indicate its status
every 1 second.

helo9 committed:
[SW] Fix bug with unresponsiveness without SD

The module became unresponsive via USB-Serial, when SD
card was removed. The issue was, that the device tried too
often to reestablish the connection with the SD card,
eating up too much computational ressources.

helo9 committed:
[SW] Implement real time clock support

The new RTCService writes clock time from RTC into log file,
e.g. 1213123, 2022-06-07 23:23:10

helo9 committed:
[SW] Format code

helo9 committed:
[SW] Add doc strings for each header file

helo9 committed:
[SW] Remove outdated python sources
